### PR TITLE
ONNX export test - BF16 support

### PR DESCRIPTION
**Pull request by galagam**
Adding support for BF16 in ONNX export
* "fake bf16" are bf16 subgraphs with fp32 I/O for testing TE vs ORT
* standard bf16 - for exporting the inputs and outputs and comparing against TRT
* added cast to bf16 before GEMM in BF16, in case input is output of DQ

### CODE REVIEW

**Comment by galagam** [comment]
@asfiyab-nvidia this is partly based on asfiyab/dev-bf16-tests, please review too

File: tests/pytorch/test_onnx_export.py [hunk 1]
```diff
@@ -157,8 +161,7 @@ def te_infer(model: torch.nn.Module, inps: Union[Tuple[torch.tensor], torch.tens
         te_outputs = model(*inps if isinstance(inps, tuple) else (inps,))
         if not isinstance(te_outputs, tuple):
             te_outputs = (te_outputs,)
-        te_outputs_np = [to_numpy(te_output) for te_output in te_outputs]
-        return te_outputs_np
+        return te_outputs
```
Comment (timmoon10) [inline tests/pytorch/test_onnx_export.py:164]: We should update the `te_infer` docstring so it no longer says:
```
Return results after copying to the CPU and converting to numpy.
```

**Review by timmoon10 — APPROVED** [review]
Overall LGTM, pending CI. My suggestions are stylistic.

File: tests/pytorch/test_onnx_export.py [hunk 2]
```diff
@@ -186,6 +191,33 @@ def compare_outputs(onnx_outputs, te_outputs, atol, rtol, max_errors_printed, al
             if nb_errors > allow_cnt_errors:
                 raise ValueError(f"Output validation of {fname} failed with {nb_errors} errors")
 
+def serialize_inputs_outputs(
+    fname: str,
+    inputs: Union[Tuple[torch.Tensor], torch.Tensor],
+    te_outputs: List[torch.Tensor],
+    input_names: List[str]=None,
+    output_names: List[str]=None,
```
Comment (ksivaman) [inline tests/pytorch/test_onnx_export.py:199]:
```suggestion
    input_names: Optional[List[str]] = None,
    output_names: Optional[List[str]] = None,
```

**Comment by ksivaman** [comment]
Pipeline 8542181

**Pull request merged**